src/post/solution.rs

```diff
@@ -0,0 +1,120 @@
+use super::{Like, PostStatus, SponsorshipToken};
+use crate::str_serializers::*;
+use crate::{AttestationId, CommentId, SolutionId, SponsorshipId};
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::serde::{Deserialize, Serialize};
+use near_sdk::{AccountId, Balance, Timestamp};
+use std::collections::HashSet;
+
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, Clone)]
+#[serde(crate = "near_sdk::serde")]
+pub struct SolutionV0 {
+    // Common fields
+    pub id: SolutionId,
+    pub name: String,
+    pub description: String,
+    pub author_id: AccountId,
+    #[serde(with = "u64_dec_format")]
+    pub timestamp: Timestamp,
+    pub status: PostStatus,
+    pub likes: HashSet<Like>,
+    pub comments: Vec<CommentId>,
+
+    // Specific fields
+    #[serde(with = "u64_dec_format")]
+    pub idea_id: u64,
+    pub attestations: Vec<AttestationId>,
+    pub sponsorships: Vec<SponsorshipId>,
+}
+
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, Clone)]
+#[serde(crate = "near_sdk::serde")]
+pub struct SolutionV1 {
+    pub name: String,
+    pub description: String,
+}
+
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, Clone)]
+#[serde(crate = "near_sdk::serde")]
+pub struct SolutionV2 {
+    pub name: String,
+    pub description: String,
+    pub requested_sponsor: Option<AccountId>,
+    #[serde(with = "u128_dec_format")]
+    pub requested_sponsorship_amount: Balance,
+    pub sponsorship_token: Option<SponsorshipToken>,
```

Comment:
Just to be aligned with other `requested_*` fields, I believe this fields should also start with `requested_`:

```suggestion
    pub requested_sponsorship_token: Option<SponsorshipToken>,
```

Reply:
Can we leave it as it is? Because otherwise it'll require additional changes on frontend

Reply:
@carina-akaia That would be unfortunate and not fixable in the future. Is it more complicated than just replacing one string with a new one in all the files?

Reply:
@frol little bit more complicated, but okay, I'll do it